### PR TITLE
Pull database images in parallel with compiling on Travis

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "build parallelization".

Proposed changes include:
* while we compiling, we can also pull database images.
* rabbitmq is not handled. Planning to do it after moving code for ldap, redis and rabbit into travis-setup-db as a separate step
* Do magic in a separate `tools/travis-pull-db.sh`, because doing it in setup-db would make that script even more unfriendly.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7006.1](https://travis-ci.org/esl/MongooseIM/jobs/571358723) / Erlang 22.0 / small_tests / 2828baca6777a4f8461a9d10509c49cf87da7b60
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2418/7006/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2418/7006/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-08-13_15.01.51/lib.mongooseim.logs/run.2019-08-13_15.01.52/suite.log.html)

File: tools/travis-pull-db.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# Environment variable DB is used by this script.
+# If DB is undefined, than this script does nothing.
```
Comment (NelsonVides) [inline tools/travis-pull-db.sh]: `than` should be `then` -> "If DB is undefined, then this script does nothing."

File: .travis.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ addons:
         # used in MUC + s2s tests combination
         - muc.localhost
 before_install:
+        - nohup ./tools/travis-pull-db.sh > travis-pull-db.log 2>&1 < /dev/null &
```
Comment (NelsonVides) [inline .travis.yml]: Nice, redirect all three I/O streams to avoid hanging :D

File: tools/travis-pull-db.sh (hunk 1 above)
Comment (arcusfelis) [reply tools/travis-pull-db.sh]: dnoe :)

**Comment by arcusfelis** [comment]
@NelsonVides this change does not have meaning for circle-ci, because:
- travis builds and pulls in parallel in one job
- circle builds in one job, caches it and spawns several new jobs, install database and run tests. Pulling and building cannot be done in parallel.

**Comment by mongoose-im** [comment]
[7011.1](https://travis-ci.org/esl/MongooseIM/jobs/571856684) / Erlang 22.0 / small_tests / 1799392645533dc03ad64eb1f66c7e5a0da130f5
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2418/7011/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2418/7011/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-08-14_14.31.58/lib.mongooseim.logs/run.2019-08-14_14.31.58/suite.log.html)

**Pull request merged**